### DISCUSSION TITLE
Simplest upgrade command for formula and greedy casks...

### DISCUSSION THREAD

**Discussion by johnalanwoods** [Homebrew/discussions#1009] [Getting started] [answered]
I'm upgrading old scripts where I used to do stuff like:

`brew upgrade --cask --greedy`

but after reading the docs it *seems* that now, `brew upgrade` takes care of casks and formula.

Great, but I also need greedy for some casks, so can I now change my scripts to simply be:

`brew update; brew upgrade --greedy;` and expect that brew will upgrade all formula AND casks greedily? 

or do I need

`brew update; brew upgrade; brew upgrade --cask --greedy;` still?

**Comment by SMillerDev** [comment, marked as answer]
No, `brew upgrade --greedy` is enough since there is no way for that to be ambiguous. (formula can't auto-update and are therefore never `--greedy`)

**Reply by infa-jowoods** [reply]
Thank you!
